test/src/unittests/tonal/test_pitchyinprobabilistic.py

```diff
@@ -0,0 +1,137 @@
+        self.assertComputeFails(PitchYinProbabilistic(), [])
+
+    def testZero(self):
+        pitch, voicedProbs = PitchYinProbabilistic()(zeros(2048))
+        self.assertEqual(pitch, -61.735)
+        self.assertEqual(voicedProbs, 0)
+
+    def testSine(self):
+        sr = 44100
+        size = sr*1;
+        freq = 440
+        signal = [sin(2.0*pi*freq*i/sr) for i in range(size)]
+        self.runTest(signal, sr, freq)
+
+    def testBandLimitedSquare(self):
+        sr = 44100
+        size = sr*1;
+        freq = 660
+        w = 2.0*pi*freq
+        nharms = 10
+        signal = zeros(size)
+        for i in range(size):
+            for harm in range(nharms):
+                signal[i] += .5/(2.*harm+1)*sin((2*harm+1)*i*w/sr)
+
+        self.runTest(signal, sr, freq)
+
+    def testBandLimitedSaw(self):
+        sr = 44100
+        size = sr*1;
+        freq = 660
+        w = 2.0*pi*freq
+        nharms = 10
+        signal = zeros(size)
+        for i in range(1,size):
+            for harm in range(1,nharms+1):
+                signal[i] += 1./harm*sin(harm*i*w/sr)
+        self.runTest(signal, sr, freq, 1.1, 0.1)
+
+    def testBandLimitedSawMasked(self):
+        sr = 44100
+        size = sr*1;
+        freq = 440
+        w = 2.0*pi*freq
+        subw = 2.0*pi*(freq-100)
+        nharms = 10
+        signal = zeros(size)
+        for i in range(1,size):
+            # masking noise:
+            whitenoise = 2*(random.rand(1)-0.5)
+            signal[i] += 2*whitenoise
+            for harm in range(1,nharms):
+                signal[i] += 1./harm*sin(i*harm*w/sr)
+        signal = 5*LowPass()(signal)
+        for i in range(1,size):
+            for harm in range(1,nharms+1):
+                signal[i] += .1/harm*sin(i*harm*w/sr)
+            signal[i] += 0.5*sin(i*subw/sr)
+        max_signal = max(signal) + 1
+        signal = signal/max_signal
+        self.runTest(signal, sr, freq, 1.5, 0.3)
+
+    def runTest(self, signal, sr, freq, pitch_precision = 1, vp_precision = 0.1):
+        frameSize = 1024
+        hopsize = frameSize
+
+        frames = FrameGenerator(signal, frameSize=frameSize, hopSize=hopsize)
+        pitchDetect = PitchYinProbabilistic(frameSize=frameSize, sampleRate = sr)
+        pitch = []
+        voicedProbs = []
+        for frame in frames:
+            f, vp = pitchDetect(frame)
+            pitch += [f]
+            voicedProbs += [vp]
+        self.assertAlmostEqual(mean(f), freq, pitch_precision)
+        # self.assertAlmostEqual(mean(voicedProbs), 0.6, vp_precision)
+
+    def testInvalidParam(self):
+        self.assertConfigureFails(PitchYinProbabilistic(), {'frameSize' : 1})
+        self.assertConfigureFails(PitchYinProbabilistic(), {'sampleRate' : 0})
+
+    # TODO: generate pitchyin/pitch_mozart_c_major_30sec.txt 
+    #       check if estimations actually have some sense
+    #       it is better to have real case using a monophonic audio 
+
+    # def testARealCase(self):
```

Comment:
Add a real regression test (given an audio input check if the resulting pitch values are as expected).

Reply:
added

Reply:
Great! I have changed waf for a flac file (to save some space).